design: Polish the table toolbars

ericpgreen2 opened the pull request:
This PR:
- Adds a common `toolbar` button type
- Adds icons to toolbar buttons
- Moves the "Flat" vs "Nested" pivot table button to the Pivot toolbar
- Moves common toolbar code into a `features/dashboards/toolbars` directory
- Improves the feel of the Search toolbar button
- Fixes the flickering of the TDD toolbar when adding a dimension filter

Dimension Table toolbar:
<img width="749" alt="image" src="https://github.com/user-attachments/assets/0eb2f366-8975-4604-8e7e-88f3c6ec3646" />

Time Dimension Detail toolbar:
<img width="1349" alt="image" src="https://github.com/user-attachments/assets/a47a881c-c11c-43b4-b055-9c3568993928" />

Pivot toolbar:
<img width="1104" alt="image" src="https://github.com/user-attachments/assets/9b5ea4bf-139b-4f81-bbef-0f126bb7265c" />


**Checklist:**
- [x] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [x] Checked if the docs need to be updated
- [ ] Intend to cherry-pick into the release branch
- [x] I'm proud of this work!

lovincyrus reviewed (changes requested):
The icons are looking good! A couple of notes:

1. There seems to be an unnecessary height in the toolbar. Can remove the h-9.

![CleanShot 2025-04-16 at 22 49 18@2x](https://github.com/user-attachments/assets/0f5b5889-8726-4aab-b5f2-5b7564221983)

2. Could be out of scope, now that the toolbar buttons' hover styles are very distinctive, you can see the triple dot icon target area is relatively smaller.

![CleanShot 2025-04-16 at 22 51 17@2x](https://github.com/user-attachments/assets/9189b8cc-7322-4a95-91c1-0090c16e0420)

ericpgreen2 commented:
> 1. There seems to be an unnecessary height in the toolbar. Can remove the h-9.

I removed the `h-9` and I also decreased the Dimension Table Header's fixed height from 50px to 40px. This arbitrary fixed height feels wrong, but removing it entirely causes the toolbar+table to feel crowded. It's out-of-scope for now, but maybe we could give the Dimension Table a defined border in conjunction with a tighter toolbar.

> 2. Could be out of scope, now that the toolbar buttons' hover styles are very distinctive, you can see the triple dot icon target area is relatively smaller.

Agree. I removed the `compact` prop on that button.

lovincyrus reviewed (approved):
lgtm
